src/ruby/puppetserver-lib/puppet/server/master.rb

```diff
@@ -97,7 +97,10 @@ def handleRequest(request)
   end
 
   def compileCatalog(request_data)
-    Puppet.override(lookup_key_recorder: create_recorder) do
+    Puppet.override(
+      lookup_key_recorder: create_recorder,
+      rich_data: Puppet[:rich_data]
```

Comment:
Should probably leave a comment here explaining why this works in puppet but here we have to explicitly set it here. If I looked at just the code it would boggle my mind as to why we are calling the override function and giving it Puppet's value.

Reply:
updated